pkg/controller/jivavolume/jivavolume_controller.go

```diff
@@ -0,0 +1,709 @@
+				WithCapacity(fmt.Sprint(capacity)),
+		).Build()
+
+	if err != nil {
+		return fmt.Errorf("failed to build statefulset object, err: %v", err)
+	}
+
+	found := &appsv1.StatefulSet{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: stsObj.Name, Namespace: stsObj.Namespace}, found)
+	if err != nil && errors.IsNotFound(err) {
+		// Set JivaVolume instance as the owner and controller
+		if err := controllerutil.SetControllerReference(cr, stsObj, r.scheme); err != nil {
+			return err
+		}
+
+		reqLog.V(2).Info("Creating a new Statefulset", "Statefulset.Namespace", stsObj.Namespace, "Sts.Name", stsObj.Name)
+		err = r.client.Create(context.TODO(), stsObj)
+		if err != nil {
+			return err
+		}
+		// Statefulset created successfully - don't requeue
+		return nil
+	} else if err != nil {
+		return operr.Wrapf(err, "failed to get the statefulset details: %v", stsObj.Name)
+	}
+
+	return nil
+}
+
+func updateJivaVolumeWithServiceInfo(r *ReconcileJivaVolume, cr *jv.JivaVolume, reqLog logr.Logger) error {
+	ctrlSVC := &v1.Service{}
+	if err := r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Name:      cr.Name + "-jiva-ctrl-svc",
+			Namespace: cr.Namespace,
+		}, ctrlSVC); err != nil {
+		return fmt.Errorf("%s, err: %v", updateErrMsg, err)
+	}
+	cr.Spec.ISCSISpec.TargetIP = ctrlSVC.Spec.ClusterIP
+	var found bool
+	for _, port := range ctrlSVC.Spec.Ports {
+		if port.Name == "iscsi" {
+			found = true
+			cr.Spec.ISCSISpec.TargetPort = port.Port
+			cr.Spec.ISCSISpec.Iqn = "iqn.2016-09.com.openebs.jiva" + ":" + cr.Spec.PV
+		}
+	}
+
+	if !found {
+		return fmt.Errorf("%s, err: can't find targetPort in target service spec: {%+v}", updateErrMsg, ctrlSVC)
+	}
+
+	reqLog.V(2).Info("Updating JivaVolume with iscsi spec", "ISCSISpec", cr.Spec.ISCSISpec)
+	cr.Status.Phase = jv.JivaVolumePhasePending
+	if err := r.client.Update(context.TODO(), cr); err != nil {
+		return fmt.Errorf("%s, err: %v, JivaVolume CR: {%+v}", updateErrMsg, err, cr)
+	}
+
+	if err := r.getJivaVolume(cr); err != nil {
+		return fmt.Errorf("%s, err: %v, JivaVolume CR: {%+v}", updateErrMsg, err, cr)
+	}
+
+	return nil
+}
+
+func createControllerService(r *ReconcileJivaVolume, cr *jv.JivaVolume,
+	reqLog logr.Logger) error {
+
+	svcObj, err := svc.NewBuilder().
+		WithName(cr.Name + "-jiva-ctrl-svc").
+		WithLabelsNew(defaultServiceLabels(cr.Spec.PV)).
+		WithNamespace(cr.Namespace).
+		WithSelectorsNew(map[string]string{
+			"openebs.io/cas-type":          "jiva",
+			"openebs.io/persistent-volume": cr.Spec.PV,
+		}).
+		WithPorts([]corev1.ServicePort{
+			{
+				Name:       "iscsi",
+				Port:       3260,
+				Protocol:   "TCP",
+				TargetPort: intstr.IntOrString{IntVal: 3260},
+			},
+			{
+				Name:       "api",
+				Port:       9501,
+				Protocol:   "TCP",
+				TargetPort: intstr.IntOrString{IntVal: 9501},
+			},
+			{
+				Name:       "m-exporter",
+				Port:       9500,
+				Protocol:   "TCP",
+				TargetPort: intstr.IntOrString{IntVal: 9500},
+			},
+		}).
```

Comment:
Please put a comment over here that ServiceType is by default set to clusterIP.

Reply:
done